src/components/Settings/AddUser.tsx

```diff
@@ -0,0 +1,137 @@
+import Locale from "@/locales";
+import { Dispatch, useState, useEffect } from "react";
+import { InputField } from "./InputField";
+import { User } from "@/shared/Classes";
+import { api } from "@/utils/api";
+
+export function AddUser({
+  t,
+  pos,
+  users,
+  setUsers,
+  editState,
+  setEditState,
+}: {
+  t: Locale;
+  pos: number;
+  users: User[];
+  setUsers: Dispatch<User[]>;
+  editState: undefined | string;
+  setEditState: Dispatch<undefined | string>;
+}) {
+  const defaultUser = new User(undefined, "", "", "", "");
+  
+  const [user, setUser] = useState(users[pos]);
+
+  const setUserMutation = api.exhibitor.setUsers.useMutation();
+  const deleteUserMutation = api.exhibitor.deleteUser.useMutation();
+
+  function handleSubmission(e: any) {
+    e.preventDefault();
+    setUserMutation.mutate({
+      id: user.id,
+      name: user.name,
+      email: user.email,
+      phone: user.phone,
+      role: user.role,
+      locale: t.locale,
+    });
+    setEditState(undefined);
+  }
+
+  function deleteUserInDatabase() {
+    deleteUserMutation.mutate({ id: users[pos].id, locale: t.locale });
+    setEditState(undefined);
+  }
+
+  useEffect(() => {
+    if (pos < users.length) setUser(users[pos]);
+  }, [users, pos]);
+
+  useEffect(() => {
+    if (setUserMutation.data) {
+      if (!setUserMutation.data.ok) {
+        setUsers([defaultUser, ...users.slice(1)]);
+        alert(setUserMutation.data.error);
+      } else if (setUserMutation.data.ok && setUserMutation.data.update)
+        setUsers(
+          users.map((u, i) => (i == 0 ? defaultUser : i == pos ? user : u))
+        );
+      else
+        setUsers([
+          ...users.map((u, i) => (i == 0 ? defaultUser : u)),
+          { ...user, id: setUserMutation.data.id },
+        ]);
+      setUserMutation.reset();
+    }
+
+    if (deleteUserMutation.data) {
+      if (!deleteUserMutation.data.ok) {
+        setUsers([defaultUser, ...users.slice(1)]);
+        alert(deleteUserMutation.data.error);
```

Comment:
![image](https://github.com/datasektionen/ddagen/assets/32957299/1deae19c-20e0-42cc-b599-21bb409e57db)

Seems like a [bug in react](https://github.com/facebook/react/issues/17355) that happens after calling `alert`.

But even without the bug, i think it would be nicer to show a red text below the button.